src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -820,8 +820,7 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						reflectionContext.AnalyzingPattern ();
 
 						var parameters = calledMethod.Parameters;
-						if ((parameters.Count == 3 && (methodParams[2].Kind == ValueNodeKind.MethodReturn || methodParams[2].Kind == ValueNodeKind.ConstInt || methodParams[2].Kind == ValueNodeKind.LoadField)
-							&& (methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0)) ||
+						if ((parameters.Count == 3 && parameters[2].ParameterType.IsTypeOf ("System", "Boolean") && (methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0)) ||
```

Comment:
Nit: check for primitive types can be done more efficiently

```suggestion
						if ((parameters.Count == 3 && parameters[2].ParameterType.MetadataType == MetadataType.Boolean && (methodParams[2].AsConstInt () == null || methodParams[2].AsConstInt () != 0)) ||
```